executor.go

```diff
@@ -52,10 +53,34 @@ func newExecutor(cfg *executorConfig) *executor {
 func (s *executor) run(mainCtx context.Context,
 	statusChan chan<- SwapInfo) error {
 
-	blockEpochChan, blockErrorChan, err :=
-		s.lnd.ChainNotifier.RegisterBlockEpochNtfn(mainCtx)
-	if err != nil {
-		return err
+	var (
+		err            error
+		blockEpochChan <-chan int32
+		blockErrorChan <-chan error
+	)
+
+	for {
+		blockEpochChan, blockErrorChan, err =
+			s.lnd.ChainNotifier.RegisterBlockEpochNtfn(mainCtx)
+		if err != nil {
+			if strings.Contains(err.Error(),
+				"in the process of starting") {
+
+				// Give chain notifier some time to start and
+				// try to re-attempt block epoch subscription.
+				select {
+				case <-time.After(50 * time.Millisecond):
+					continue
```

Comment:
Perhaps add a log here that we're waiting for the chain notifier? Otherwise loop will just be hanging and user may not know why?

Reply:
Logging sounds good to me. And maybe increase the delay to 500ms to not spam `lnd` too much? It's got enough on its plate at this moment of startup anyway :joy:

Reply:
Yeah the 50ms was a bit overkill :) Log added.